### ISSUE TITLE
Lazy relations and Query Builder join

### ISSUE THREAD

**Issue by dlindenkreuz** [typeorm/typeorm#996]
When using the query builder's `leftJoinAndSelect` with lazy relations (defined with `Promise<Entity>`), the relation does not get added to the result as a "regular" property. The actual data can be found at the double-underscore key (ex. `__room__` instead of `room`). The async accessor is still attached.

This means that accessing a lazy relation that was fetched eagerly using `leftJoinAndSelect` will actually fetch the data twice. This leads to a lot of unnecessary queries when the result list is already long.

```ts
const x = await getRepository(Share)
    .createQueryBuilder("share")
    .leftJoinAndSelect("share.room", "room")
    .getMany()
// → SELECT [...etc.] FROM "share" "share" LEFT JOIN "room" "room" ON "room"."id"="share"."roomId"
console.log(x[0])
// → Share { id: 1, [...etc.]  __room__: Room { [...etc.] } }
console.log(x[0].room)
// → Promise
console.log(await x[0].room)
// → SELECT [...etc.] FROM "room" "room" INNER JOIN "share" "Share" ON "Share"."roomId" = "room"."id" WHERE "room"."id" = $1
```

**Expected behaviour:**
When a relation is defined as lazy and is fetched eagerly using the query builder, existing data should be used and no additional SQL query should be made.

**Typeorm version:** 0.1.0-alpha.50

**Comment by luchillo17** [comment]
So it should check if data for such relation is already loaded, and only if not then fetch it?

**Comment by dlindenkreuz** [comment]
Exactly.

**Comment by luchillo17** [comment]
Doesn't seem that hard to do but i've had a hard time with the project folder structure and the architecture as well, if nobody tackles this up to next week i may try.

**Comment by pleerock** [comment]
Will be very nice if you can help on it @luchillo17

**Comment by luchillo17** [comment]
I'm having a bad week, lots of things to do, i may take a look at it sunday.

**Comment by pleerock** [comment]
Awesome! Thanks

**Comment by luchillo17** [comment]
Sorry, this week busy too, hope to finish my project for this week.

**Comment by pleerock** [comment]
I've fixed this issue and it will be released in `0.1.6`
